netty/src/main/java/io/grpc/netty/NettyServerHandler.java

```diff
@@ -383,6 +385,20 @@ private void onHeadersRead(ChannelHandlerContext ctx, int streamId, Http2Headers
         return;
       }
 
+      if (headers.authority() == null) {
```

Comment:
I was wondering if you can also have a case of multiple authority headers. This comment https://github.com/grpc/grpc-go/issues/3444#issuecomment-614875030 also confirmed my suspicion. But may be this PR is not for checking that.

Reply:
The current code structure isn't great for that. I thought it wasn't all that possible without changing Netty, but looking again found a way. I've added a new commit that adds the validation, but it continues the hack that throws a checked exception from a method that doesn't throw checked exceptions, it uses a connection error instead of stream error, and is not practically testable.